src/main/java/io/micronaut/starter/analytics/postgres/FeatureRepository.java

```diff
@@ -106,4 +114,60 @@ private List<TotalDTO> resultSetToTotals(ResultSet resultSet) throws SQLExceptio
     private String query(String name, String table) {
         return "SELECT " + name + " AS name, count(*) AS total FROM " + table + " GROUP BY name ORDER BY total";
     }
+
+    @ReadOnly
+    List<PercentageDTO> applicationTypePercentages() {
+        return getPercentages(FIELD_APPLICATION_TYPE);
+    }
+
+    @ReadOnly
+    List<PercentageDTO> buildToolPercentages() {
+        return getPercentages(FIELD_BUILD_TOOL);
+    }
+
+    @ReadOnly
+    List<PercentageDTO> jdkPercentages() {
+        return getPercentages(FIELD_JDK_VERSION);
+    }
+
+    @ReadOnly
+    List<PercentageDTO> languagePercentages() {
+        return getPercentages(FIELD_LANGUAGE);
+    }
+
+    @ReadOnly
+    List<PercentageDTO> testFrameworkPercentages() {
+        return getPercentages(FIELD_TEST_FRAMEWORK);
+    }
+
+    private List<PercentageDTO> getPercentages(String name) {
+        return this.jdbcOperations
+                .prepareStatement(percentageQuery(name), statement -> {
+                    try (ResultSet resultSet = statement.executeQuery()) {
+                        List<PercentageDTO> results = new ArrayList<>(40);
+                        while (resultSet.next()) {
+                            results.add(
+                                    new PercentageDTO(
+                                            resultSet.getString("name"),
+                                            resultSet.getDouble("percentage")
+                                    )
+                            );
+                        }
+                        return results;
+                    }
+                });
+    }
+
+    private String percentageQuery(String name) {
+        return """
+                WITH totals AS (
+                    SELECT %s AS name, count(%s) AS total FROM application GROUP BY name
+                ),
+                summed_totals AS (
+                    SELECT SUM(total) AS total FROM totals
+                )
+                SELECT totals.name, ROUND(totals.total / summed_totals.total, 4) AS percentage
```

Comment:
is it best to to this math at the database layer?

Reply:
Databases can do maths 😜

Is there a reason for doing in the application layer?

Reply:
I can switch it if you'd prefer?

Reply:
> Is there a reason for doing in the application layer?

Yes, I prefer to keep as much logic in application code vs persistence code when possible. but let's keep it this way.